src/app/api/common/delegations/getDelegations.ts

```diff
@@ -122,97 +92,175 @@ async function getCurrentDelegateesForAddress({
+              contracts.token.address.toLowerCase(),
+              skip,
+              take
+            );
+          },
+          page,
+          pageSize
+        );
+      })(),
+      contracts.token.provider.getBlock("latest"),
+    ]
+  );
 
-  // KENT: Commented out Direct delegations, needs to be paginated and optimized for prod
-  // const directDelegators = (async () => {
-  //   const proxyAddress = await getProxyAddress(address);
-  //   if (!proxyAddress) {
-  //     return [];
-  //   }
-  //   return prisma.$queryRaw<Prisma.DelegateesGetPayload<true>[]>(
-  //     Prisma.sql`
-  //     SELECT *
-  //     FROM (
-  //     SELECT
-  //       delegator,
-  //       delegatee,
-  //       block_number
-  //     FROM (
-  //       SELECT
-  //           delegator,
-  //           to_delegate as delegatee,
-  //           block_number,
-  //           ROW_NUMBER() OVER (PARTITION BY delegator ORDER BY block_number DESC, log_index DESC, transaction_index DESC) as rn
-  //       FROM optimism.delegate_changed_events
-  //       WHERE to_delegate=${address.toLowerCase()}
-  //     ) t1
-  //     WHERE rn=1
-  //     ) t2
-  //     LEFT JOIN LATERAL (
-  //       SELECT
-  //         COALESCE(SUM(
-  //           CASE WHEN "from"=delegator THEN -"value"::NUMERIC ELSE "value"::NUMERIC END
-  //         ), 0) AS balance
-  //       FROM optimism.transfer_events
-  //       WHERE "from"=delegator OR "to"=delegator
-  //     ) t3 ON TRUE
-  //     `
-  //   );
-  // })();
+  return {
+    meta: directDelegators.meta,
+    data: [
+      ...advancedDelegators.map((advancedDelegator) => ({
+        from: advancedDelegator.from,
+        to: advancedDelegator.to,
+        allowance: advancedDelegator.delegated_amount.toFixed(0),
+        timestamp: latestBlock
+          ? getHumanBlockTime(advancedDelegator.block_number, latestBlock)
+          : null,
+        type: "ADVANCED" as const,
+        amount:
+          Number(advancedDelegator.delegated_share.toFixed(3)) === 1
+            ? ("FULL" as const)
+            : ("PARTIAL" as const),
+        transaction_hash: advancedDelegator.transaction_hash || "",
+      })),
+      ...(
+        await Promise.all(
+          directDelegators.data.map(async (directDelegator) => ({
+            from: directDelegator.delegator,
+            to: directDelegator.delegatee,
+            allowance: (
+              await contracts.token.contract.balanceOf(
+                directDelegator.delegator
+              )
+            ).toString(),
+            timestamp: latestBlock
+              ? getHumanBlockTime(directDelegator.block_number, latestBlock)
+              : null,
+            type: "DIRECT" as const,
+            amount: "FULL" as const,
+            transaction_hash: directDelegator.transaction_hash,
+          }))
+        )
+      ).filter((delegator) => BigInt(delegator.allowance) > BigInt(1e15)), // filter out delegators with 0 (or close to 0) balance
+    ],
+  };
+}
```

Comment:
Is this more readable and faster?  I assume doing the filter before the `getHumanBlockTime`

``` typescript
const advancedDelegatorsData = advancedDelegators.map((advancedDelegator) => ({
  from: advancedDelegator.from,
  to: advancedDelegator.to,
  allowance: advancedDelegator.delegated_amount.toFixed(0),
  timestamp: latestBlock
    ? getHumanBlockTime(advancedDelegator.block_number, latestBlock)
    : null,
  type: "ADVANCED" as const,
  amount:
    Number(advancedDelegator.delegated_share.toFixed(3)) === 1
      ? ("FULL" as const)
      : ("PARTIAL" as const),
  transaction_hash: advancedDelegator.transaction_hash || "",
}));

const directDelegatorsData = await Promise.all(
  directDelegators.data.map(async (directDelegator) => ({
    from: directDelegator.delegator,
    to: directDelegator.delegatee,
    allowance: (
      await contracts.token.contract.balanceOf(directDelegator.delegator)
    ).toString(),
    timestamp: latestBlock
      ? getHumanBlockTime(directDelegator.block_number, latestBlock)
      : null,
    type: "DIRECT" as const,
    amount: "FULL" as const,
    transaction_hash: directDelegator.transaction_hash,
  }))
);

const filteredDirectDelegatorsData = directDelegatorsData.filter(
  (delegator) => BigInt(delegator.allowance) > BigInt(1e15) // filter out delegators with 0 (or close to 0) balance
);

return {
  meta: directDelegators.meta,
  data: [...advancedDelegatorsData, ...filteredDirectDelegatorsData],
};
```

Reply:
yes ty!